activemodel/lib/active_model/type/collection.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module ActiveModel
+  module Type
+    # Attribute type for a collection of values. It is registered under
+    # the +:collection+ key. +:element_type+ option is used to specify elements type
+    #
+    #   class User
+    #     include ActiveModel::Attributes
+    #
+    #     attribute :lucky_numbers, :collection, element_type: :integer
+    #   end
+    #
+    #   user = User.new(lucky_numbers: [1, 2, 3])
+    #   user.lucky_numbers # => [1, 2, 3]
+    #
+    # Value is wrapped into an Array if not an Array already
+    #
+    #   User.new(lucky_numbers: 1).lucky_numbers # => [1]
+    #
+    # Collection elements are coerced by their +:element_type+ type
+    #
+    #   User.new(lucky_numbers: ["1"]).lucky_numbers # => [1]
+    #
+    class Collection < Value
+      def initialize(**args)
+        @element_type = args.delete(:element_type)
+        @type_object = Type.lookup(element_type)
+        @serializer = args.delete(:serializer) || ActiveSupport::JSON
+        super
+      end
+
+      def type
+        :collection
+      end
+
+      def cast(value)
+        return [] unless value
```

Comment:
If I understand correctly, this would cause `true` and `false` to be handled differently, no?  e.g.:

```ruby
model.my_boolean_collection = true
model.my_boolean_collection # => [true]
model.my_boolean_collection = false
model.my_boolean_collection # => []
```

So, perhaps:

```suggestion
        return [] if value.nil?
```

(Note that `Array(nil)` evaluates to `[]`, so we could also just remove this line, but checking `nil?` could save an array allocation from the `map`.)

Reply:
That's a very good catch, you are right! Let me change the method and add more tests for this case
Thanks!